src/test/java/org/openrewrite/java/migrate/javax/UseJoinColumnForMappingTest.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate.javax;
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.DocumentExample;
+import org.openrewrite.InMemoryExecutionContext;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+class UseJoinColumnForMappingTest implements RewriteTest {
```

Comment:
The following these cases should be ported over from our bucket:

- @OneToOne used with @Column (with a few attributes provided)
- @ManyToOne and @Column, both without any attributes
- @OneToOnee and @Column, both without any attributes

We should also test the use of these annotations within a sub class here as well.

Reply:
Did you mean subclass extending the `@Entity` class? or an inner class inside the `@Entity` class?